apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+#                       get help with the commands and descriptions.
+#
+# Version           1.0 - Initial Release
+#
+# Notes             Grab your own api key at https://apixu.com and place it in
+#                   the "apikey" variable below. Set private variable to 0 or 1
+#                   if you want the set command responses in PM.
+#
+########################################################################################
+
+namespace eval weather {
+    package require json
+    package require tdom
+    package require http
+    package require tls
+
+    # To be able to query APIs that use https
+    http::register https 443 [list tls::socket -tls1 1 -ssl2 0 -ssl3 0]
+
+    setudef flag weather-apixu
+    # Default units for weather output. 0 for metric, 1 for imperial, 2 for both.
+    variable units "1"
+    variable base_url "https://api.apixu.com/v1"
+    # Set your apikey here
+    variable apikey "<insert-your-key-here>"
+    # Set to 1 (0 by default) if you want set weather responses
+    # to use PM/notify. Set to 0 if you want it all in the channel.
+    variable private 0
+
+    # Public channel command bindings
+    bind pub - .wz weather::current
+    bind pub - .wzf weather::forecast
+    bind pub - .set weather::location
+
+    # Private message command bindings
+    bind msg - .wz weather::pm_current
+    bind msg - .wzf weather::pm_forecast
+    bind msg - .set weather::pm_location
+
+    ##########################
+    # Public command functions
+    ##########################
+    proc current {nick uhost hand chan text} {
+        set text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+            return
+        }
+
+        # Checks to see if a user has a location set
+        set userinfo [_get_userinfo $nick $hand]
+        if {$text eq "" && [dict exists $userinfo error]} {
+            if {$::weather::private} {
+                puthelp "NOTICE $nick :Did you want the weather for a specific location?\
+                         Or please PM me with the \".set\" command to set a default location."
+                return
+            }
+
+            puthelp "PRIVMSG $chan :No location set or use \002'.wz <location>'\002"
+            return
+        } elseif {[dict exists $userinfo error]} {
+            set userinfo [dict create location $text units $::weather::units]
+        } elseif {$text ne ""} {
+            set userinfo [dict create location $text units [dict get $userinfo units]]
+        }
+
+        set location [dict get $userinfo location]
+        if {[catch {set json_data [_get_json $location "current"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        if {[catch {set current_weather [_json_parse $json_data $userinfo "current"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        puthelp "PRIVMSG $chan :$current_weather"
```

Comment:
Use `putquick` here.  It's a faster queue and will help with queue contention when people are spamming the bot with weather requests.  The happy path will be more efficient, leaving more resources for the unhappy path.